tests/new_sequence.rs

```diff
@@ -48,36 +43,63 @@ fn works_with_valid_path_and_name() {
+    setup::try_make_sequence(&root_key_path, "New_Sequence", "Dissonance.ogg");
+}
 
-    let name = "New_Sequence".to_string();
+#[test]
+#[should_panic(expected = "User not found")]
+fn fails_with_no_user_private_key() {
+    let root = setup::setup_init_cd();
+    let key_path = common::make_key_file(&root.path(), "a.pem", TestKey::GoodKeyPem);
+    setup::try_make_sequence(&key_path, "New_Sequence", "Dissonance.ogg");
+}
 
-    let music_file_path = get_music_file_path("Dissonance.mp3");
+#[test]
+#[should_panic(expected = "Unauthorized action")]
+fn fails_with_no_admin_private_key() {
+    let root = setup::setup_init_cd();
+    let root_key_path = common::make_key_file(&root.path(), "root.pem", TestKey::RootKeyPem);
+    let key_path = common::make_key_file(&root.path(), "a.pem", TestKey::GoodKeyPem);
+
+    setup::try_new_user(&root_key_path, &root.path(), "Test user", "a.pub", TestKey::GoodKeyPub);
+    setup::try_make_sequence(&key_path, "New_Sequence", "Dissonance.ogg");
+}
 
-    let _ = match proton_cli::new_sequence(&name, &music_file_path) {
-        Ok(_) => (),
-        Err(e) => panic!(e.to_string()),
-    };
+#[test]
+#[should_panic(expected = "SSL error")]
+fn fails_with_invalid_private_key() {
+    let root = setup::setup_init_cd();
+    let root_key_path = common::make_key_file(&root.path(), "root.pem", TestKey::GoodKeyPub);
+    setup::try_make_sequence(&root_key_path, "New_Sequence", "Dissonance.ogg");
+}
+
+#[test]
+#[allow(unused_variables)]
+// root reference must be kept to keep temp directory in scope, but is never used
+#[should_panic(expected = "Unsupported file type")]
+fn fails_with_invalid_file_extension() {
+    let root = setup::setup_init_cd();
+    let root_key_path = common::make_key_file(&root.path(), "root.pem", TestKey::RootKeyPem);
+    setup::try_make_sequence(&root_key_path.as_path(), "New_Sequence", "Dissonance.mp3");
 }
 
 #[test]
 #[should_panic(expected = "Duplicate sequence")]
 fn fails_with_duplicate_sequence_name() {
-    let root = common::setup_init_cd();
+    let root = setup::setup_init_cd();
 
-    let name = "New_Sequence".to_string();
+    let name = "New_Sequence";
+    let root_key_path = common::make_key_file(&root.path(), "root.pem", TestKey::RootKeyPem);
 
-    let music_file_path_a = get_music_file_path("Dissonance.ogg");
-    let music_file_path_b = get_music_file_path("GlorytotheBells.ogg");
+    setup::try_make_sequence(&root_key_path.as_path(), &name, "Dissonance.ogg");
 
-    match proton_cli::new_sequence(&name, &music_file_path_a) {
-        Ok(_) => (),
-        Err(e) => panic!(e.to_string()),
-    };
+    let music_file_path = common::get_music_file_path("GlorytotheBells.ogg");
 
-    match proton_cli::new_sequence(&name, &music_file_path_b) {
+    match proton_cli::new_sequence(&root_key_path.as_path(), &name, &music_file_path.as_path()) {
         Ok(_) => (),
         Err(e) => {
             // Make sure the second music file wasn't copied
```

Comment:
This is kind of hard to express, but I think this test has the problem that you don't know if its failing where it should be failing. There's the possibility that the "Duplicate sequence" error is being thrown on line 98. It wouldn't in actuality, but it sort of looks like it could if you don't know the code.

Therefore, I think you should change it to "Duplicate sequence detected, extra music file not copied" and throw it on line 108. What do you think?